lineapy/utils/analytics/usage_tracking.py

```diff
@@ -33,6 +34,17 @@ def _py_version():
     )
 
 
+@lru_cache(maxsize=1)
+def _runtime():
+    if options.get("runtime") is not None:
+        return str(options.get("runtime"))  # Return manually set value
+
+    if get_ipython() is not None:
+        return str(get_ipython())
```

Comment:
I would prefer if we do some checking here. For instance, a quick check returned`<google.colab._shell.Shell at 0x7f8439f50190>`, which could change to something e.g., what you had `<google.colab._shell.Shell object at 0x7ff79b1d1250>`.

There are a couple options here.
1. Do something hacky---like a regex on the str
2. Do something more robust and do a class checking of the form `isinstance(get_ipython(), ....)`. But you'd have to play around with how to do the imports.

The method should also specify the return type---I would prefer it to be a fixed set of strings or enums, e.g., "colab", "jupyter", "binder", "databricks" etc.

Be explicit about what this works for and doesn't work for. E.g., did you test on dataricks?

Reply:
Another way to detect which environment is looking at which module has been pre-installed.

For Colab, you can `import colab`, for Databricks, you can `import dbutils` . Not so sure about Binder

Reply:
Thanks for the suggestions! I adapted these ideas in a new implementation. See message below.